rust_dev_preview/dynamodb/src/scenario/list.rs

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use aws_sdk_dynamodb::{Client, Error};
+use tokio_stream::StreamExt;
+
+// List your tables.
+// snippet-start:[dynamodb.rust.list-tables]
+pub async fn list_tables(client: &Client) -> Result<(), Error> {
+    let paginator = client.list_tables().into_paginator().items().send();
+    let table_names = paginator.collect::<Result<Vec<_>, _>>().await?;
+
+    println!("Tables:");
+
+    for name in &table_names {
+        println!("  {}", name);
+    }
+
+    println!("Found {} tables", table_names.len());
+    Ok(())
+}
+// snippet-end:[dynamodb.rust.list-tables]
+
+// List only 10 of your tables.
+// snippet-start:[dynamodb.rust.list10-tables]
+pub async fn list_tables_limit_10(
+    client: &aws_sdk_dynamodb::Client,
+) -> Result<(), aws_sdk_dynamodb::Error> {
+    let resp = client.list_tables().limit(10).send().await?;
+
+    println!("Tables:");
+
+    let names = resp.table_names().unwrap_or_default();
+
+    for name in names {
+        println!("  {}", name);
+    }
+
+    println!();
+    println!("Found {} tables", names.len());
+
+    Ok(())
+}
+// snippet-end:[dynamodb.rust.list10-tables]
+
+// List your tables 10 at a time.
+// snippet-start:[dynamodb.rust.list-more-tables]
+pub async fn list_tables_iterative(client: &Client) -> Result<(), Error> {
+    // snippet-start:[dynamodb.rust.list-more-tables-list]
+    let mut resp = client.list_tables().limit(10).send().await?;
+    let names = resp.table_names.unwrap_or_default();
+    let len = names.len();
+
+    let mut num_tables = len;
+
+    println!("Tables:");
+
+    for name in names {
+        println!("  {}", name);
+    }
+
+    while resp.last_evaluated_table_name != None {
+        println!("-- more --");
+        resp = client
+            .list_tables()
+            .limit(10)
+            .exclusive_start_table_name(
+                resp.last_evaluated_table_name
+                    .as_deref()
+                    .unwrap_or_default(),
+            )
+            .send()
+            .await?;
+
+        let names = resp.table_names.unwrap_or_default();
+        num_tables += names.len();
+
+        for name in names {
+            println!("  {}", name);
+        }
+    }
+
+    println!();
+    println!("Found {} tables", num_tables);
+
+    Ok(())
+    // snippet-end:[dynamodb.rust.list-more-tables-list]
+}
+// snippet-end:[dynamodb.rust.list-more-tables]
+
+// List up to 10 tables and indicates whether there are more.
+// snippet-start:[dynamodb.rust.are-more-tables]
+pub async fn list_tables_are_more(client: &Client) -> Result<(), Error> {
```

Comment:
Is there a good reason for `list_tables_limit_10` to be separate from `list_tables_are_more`? I don't understand why we should have two separate examples for this when the second one merely adds an `if` statement at the end.

Reply:
I have no idea why this sample even exists, but tracking that down is a task for another day